metrics.md

```diff
@@ -0,0 +1,226 @@
+
+## Health Metrics
+
+There are three primary health metrics that are exposed by the side-car:
+
+* [`side_car_health_check_system_updates_total`](#side_car_health_check_system_updates_total): This metric is a counter that increments every time the side-car updates its internal state. This is a good indicator of the side-car's overall health.
+* [`side_car_health_check_ticker_updates_total`](#side_car_health_check_ticker_updates_total): This metric is a counter that increments every time the side-car updates the price of a given market. This is a good indicator of the overall health of a given market.
+* [`side_car_health_check_provider_updates_total`](#side_car_health_check_provider_updates_total): This metric is a counter that increments everytime the side-car utilizes a given providers market data. This is a good indicator of the health of a given provider. Note that providers may not be responsible for every market. However, the side-car correctly tracks the number of expected updates for each provider.
+
+### `side_car_health_check_system_updates_total`
+
+This metric should be monotonically increasing. Specifically, the rate of this metric should be inversely correlated to the configured `UpdateInterval` in the oracle side-car configuration (`oracle.json`). To check this, you can run the following query in Prometheus:
+
+```promql
+rate(side_car_health_check_system_updates_total[5m])
+```
+
+![Architecture Overview](./resources/side_car_health_check_system_updates_total_rate.png)
+
+
+### `side_car_health_check_ticker_updates_total`
+
+This should be a monotonically increasing counter for each market. Each market's counter should be relatively close to the `side_car_health_check_system_updates_total` counter.
+
+To verify that the rate of updates for each market is as expected, you can run the following query in Prometheus:
+
+```promql
+rate(side_car_health_check_ticker_updates_total[5m])
+```
+
+![Architecture Overview](./resources/side_car_health_check_ticker_updates_total_rate.png)
+
+### `side_car_health_check_provider_updates_total`
+
+This metric should be monotonically increasing for each (provider, market) pair. To verify that the rate of updates for each provider is as expected, you can run the following query in Prometheus:
+
+```promql
+rate(side_car_health_check_provider_updates_total{provider="coinbase_api", success="true"}[5m])
+```
+
+![Architecture Overview](./resources/side_car_health_check_provider_updates_total_rate.png)
+
+### Health Metrics Summary
+
+In summary, the health metrics should be monitored to ensure that the side-car is updating its internal state, updating the price of each market, and fetching data from the price providers as expected. The rate of updates for each of these metrics should be inversely correlated with the `UpdateInterval` in the oracle side-car configuration.
```

Comment:
Maybe we should describe how you could know there is something wrong here.

For example,

In the case where a health check update counter is not being updated at the rate it had been over X window of time...